### DISCUSSION TITLE
URL and Path conflict?

### DISCUSSION THREAD

**Discussion by safaci2000** [earthboundkid/requests#100] [Support] [answered]
This might be intended behavior but I just wanted to make sure this is the case.

```go
	req := requests.URL(url)
	req.Header("Authorization", "Bearer secret1234")
       	req.Path(path).
		Accept(ContentTypeJSON).
		ToJSON(&responseObj).
		BodyJSON(req.CaptureRequest).
		Method(http.MethodDelete).
		Fetch(ctx)
```

if the url is something like http://localhost/api/v1/
and the path is: /users/list

the generated URL will http://localhost/users/list where it ignores the prefix of the URL and simply uses the path passed on.  Is this intentional? It seems like a bug to me, but maybe this is expected?

**Comment by earthboundkid** [comment, marked as answer]
[This is the expected behavior](https://pkg.go.dev/github.com/carlmjohnson/requests#Builder.Path). To append a path, it needs to be relative (not begin with a slash). Try “users/list” or “./users/list” instead of “/users/list”.

**Reply by safaci2000** [reply]
This feels like a pretty non-intuitive behavior.  

The path needs to not start with / and  the URL has to end with a /, if either of those conditions aren't met it won't work.

Thanks for clarifying the intent though.

**Reply by earthboundkid** [reply]
If you can improve the wording of the docs, I would love a PR. 😄
